### PR TITLE
WIP: useEvents hook

### PR DESCRIPTION
**Pull request by christophermohammed**
# Started work on Events

1. Used [reactfire](https://github.com/FirebaseExtended/reactfire) to create a firebase provider and added it to the `AppProvider` Component. 
2. Created an `Event` type consisting of `EventData` (title, start, end...) and `EventMetadata` (id, creationDate) for firebase
3. Created a `useEvents` hook that returns a list of all events in the DB and an `addEvent` function that allows users to add an event.
4. Wrote some small tests that verify the return value of the `useEvents` hook

### CODE REVIEW

File: src/components/AppProvider/AppProvider.tsx [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+import { FirebaseAppProvider } from 'reactfire';
 import { ThemeDecorator } from 'utils/theme';
+import { firebaseConfig } from 'utils/firebase';
 
 const AppProvider: React.FC = ({ children }) => {
-  return <ThemeDecorator>{children}</ThemeDecorator>;
+  return (
+    <ThemeDecorator>
+      <FirebaseAppProvider firebaseConfig={firebaseConfig}>
+        <Suspense fallback={<p>loading status...</p>}>{children}</Suspense>
```
Comment (jtaaa) [inline src/components/AppProvider/AppProvider.tsx]: Add creating a loading component or multiple loading components to the to-do list.

File: src/components/AppProvider/AppProvider.tsx [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+import React, { Suspense } from 'react';
+import { FirebaseAppProvider } from 'reactfire';
 import { ThemeDecorator } from 'utils/theme';
+import { firebaseConfig } from 'utils/firebase';
 
 const AppProvider: React.FC = ({ children }) => {
-  return <ThemeDecorator>{children}</ThemeDecorator>;
+  return (
+    <ThemeDecorator>
+      <FirebaseAppProvider firebaseConfig={firebaseConfig}>
```
Comment (jtaaa) [inline src/components/AppProvider/AppProvider.tsx]: Why is the `ThemeDecorator` wrapping the `FirebaseAppProvider`. One day we might want to use something stored in firebase to determine which theme values to use. I think the `FirebaseAppProvider` should wrap the `ThemeDecorator` instead.

**Comment by jtaaa** [comment]
You can go ahead and merge. Take note of the to-dos and you can use this PR after it's closed as a record

File: src/components/AppProvider/AppProvider.tsx (hunk 2 above)
Comment (christophermohammed) [reply src/components/AppProvider/AppProvider.tsx]: Got it. Makes sense👍

File: src/components/AppProvider/AppProvider.tsx (hunk 1 above)
Comment (christophermohammed) [reply src/components/AppProvider/AppProvider.tsx]: I created a board in `Projects` and moved the to-dos there

**Pull request merged**